src/c++/perf_analyzer/genai-perf/genai_perf/plots/base_plot.py

```diff
@@ -41,40 +38,46 @@ class BasePlot:
     Base class for plots
     """
 
-    def __init__(self, stats: Statistics, extra_data: Dict | None = None) -> None:
-        self._stats = stats
-        self._metrics_data = deepcopy(stats.metrics.data)
-        if extra_data:
-            self._metrics_data = self._metrics_data | extra_data
+    def __init__(self, data: list[ProfileRunData]) -> None:
+        self._profile_data = data
 
     def create_plot(
         self,
-        x_key: str,
-        y_key: str,
-        x_metric: str,
-        y_metric: str,
         graph_title: str,
         x_label: str,
         y_label: str,
         filename_root: str,
+        output_dir: Path,
     ) -> None:
         """
         Create plot for specific graph type
         """
         raise NotImplementedError
 
-    def _generate_parquet(self, dataframe: DataFrame, file: str) -> None:
-        dataframe.to_parquet(
-            f"{DEFAULT_ARTIFACT_DIR}/data/{file}.gzip", compression="gzip"
+    def _create_dataframe(self, x_label: str, y_label: str) -> pd.DataFrame:
+        return pd.DataFrame(
```

Comment:
We did. The format of dataframe is not ideal since dataframe is 2D by nature and our multi-run data requires higher dimensional data, but that is to be revisited later.